### PR TITLE
Added BooleanField field Type

### PR DESCRIPTION
**Pull request by rowan08**
Added the field to Field.h, Model.h and Model.cpp.
Also added to Model::insertBatch() to insert data for BOOL field types
Tested by adding a BooleanField, proteinIsDrugTarget, to test.cpp and setting the value to true. The script runs fine.
Issue: https://github.com/davidbrownza/CORMAP/issues/9

### CODE REVIEW

File: Model.cpp [hunk 1]
```diff
@@ -129,6 +137,15 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
                         conn.setString(paramNum, i->getValue());
                     }
                 }
+                else if(f->getType() == BOOLEAN)
```
Comment (davidbrownza) [inline Model.cpp]: Just a side note, we should probably convert this if statement to a switch at some point in the future. Long term, there is a repeating pattern here that we could probably refactor.

Comment (davidbrownza) [reply Model.cpp]: For a start, the null check could be done before this if statement, which would mean we wouldn't have to repeat it within each block.

Comment (davidbrownza) [reply Model.cpp]: Actually, now is as good a time as any to do this. Convert this whole if statement to a switch and perform the null check before the switch.

**Review by davidbrownza — CHANGES_REQUESTED** [review]
You did everything right, but I have added on an extra requirement :P

File: Model.cpp [hunk 2]
```diff
@@ -92,41 +100,46 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
...
-                {
-                    CharField * i = static_cast <CharField*>(f);
-                    if (i->isNull()) {
-                        conn.setNull(paramNum);
-                    } else {
-                        conn.setString(paramNum, i->getValue());
+                    switch(f->getType()){
+
+                        case INTEGER: 
+                        {
```
Comment (davidbrownza) [inline Model.cpp]: Remove these brackets in the cases, they are not necessary.

File: Model.cpp [hunk 3]
```diff
@@ -92,41 +100,46 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
...
-                        conn.setNull(paramNum);
-                    } else {
-                        conn.setInt(paramNum, i->getValue());
-                    }
-                }
-                else if(f->getType() == FLOAT)
-                {
+                if (f->isNull()) {
+                    conn.setNull(paramNum);
+                } else {
```
Comment (davidbrownza) [inline Model.cpp]: You can remove the `else` block if you use a `continue` statement in the `if` block after setting the parameter to null.

**Review by davidbrownza — CHANGES_REQUESTED** [review]
Two small changes.

File: Model.cpp (hunk 2 above)
Comment (rowan08) [reply Model.cpp]: If I remove the brackets (and nothing else), it throws errors along the lines of `error: jump to case label [-fpermissive]`,` error:   crosses initialization of ‘IntegerField* i’`  and ` error: conflicting declaration ‘BooleanField* i’ `.

This seems to have something to do with declaring variables within a switch statement, which is not allowed for reasons I don't entirely follow (see below...), but they recommend adding an additional layer of scope to fix this problem. 

https://stackoverflow.com/questions/449736/why-can-t-variables-be-declared-in-a-switch-statement

Comment (davidbrownza) [reply Model.cpp]: Does it work if you move the `break` inside the brackets:
```
case INTEGER:
{
    IntegerField * i = static_cast <IntegerField*>(f);
    conn.setInt(paramNum, i->getValue());
    break;
}
```
If not, it is fine, just fix up the indentation like so:
```
case INTEGER:
    {
        IntegerField * i = static_cast <IntegerField*>(f);
        conn.setInt(paramNum, i->getValue());
    }
    break;
```

**Review by davidbrownza — CHANGES_REQUESTED** [review]

**Review by davidbrownza — APPROVED** [review]

**Pull request merged**